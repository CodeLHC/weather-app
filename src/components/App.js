import "../styles/App.css";
import React, { useState } from "react";
import getForecast from "../requests/getForecast";
import LocationDetails from "./LocationDetails";
import ForecastList from "./ForecastList";
import ForecastDetails from "./ForecastDetails";

function App() {
  const [location, setLocation] = useState({
    city: "",
    country: "",
  });
  const [forecasts, setForecasts] = useState([]);
  const [selectedDate, setSelectedDate] = useState(0);
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const updateForecast = async () => {
    const forecastData = await getForecast();
    setLocation(forecastData.location);
    setForecasts(forecastData.forecasts);
    setSelectedDate(forecastData.forecasts[0].date);
  };

  React.useEffect(() => {
    updateForecast();
  }, []);

  return (
    <div className="weather-app">
      <h1>Weather App</h1>
      <LocationDetails city={location.city} country={location.country} />
      <ForecastList
        forecasts={forecasts}
        handleForecastSelect={handleForecastSelect}
      />

      {selectedDate && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
}

export default App;
