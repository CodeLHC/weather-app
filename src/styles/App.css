.weather-app {

    margin: 30px;
}
.heading {
    display: flex;
    justify-content: center;
}

h1 {
   color:#2B2D42;
   font-size: 45px;
}
