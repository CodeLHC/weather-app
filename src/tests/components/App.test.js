import { render } from "@testing-library/react";
import React from "react";
import App from "../../components/App";

describe("App", () => {
  const location = {
    city: "Manchester",
    country: "GB",
  };
  const validProps = [
    {
      date: 1111111,
      description: "Stub description 1",
      icon: "stubIcon1",
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: 2222222,
      description: "Stub description2",
      icon: "stubIcon2",
      temperature: {
        max: 24,
        min: 13,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <App location={location.location} forecasts={validProps.forecasts} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
