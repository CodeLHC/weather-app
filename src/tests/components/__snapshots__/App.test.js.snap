// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Weather App
    </h1>
    <div>
      <h1
        class="locationInfo"
      >
        , 
      </h1>
    </div>
    <div
      class="forecast-list"
    />
    0
  </div>
</DocumentFragment>
`;
